miqa/celery.py

```diff
@@ -14,3 +16,11 @@
 
 # Load task modules from all registered Django app configs.
 app.autodiscover_tasks()
+
+if settings.DEMO_MODE:
+    app.conf.beat_schedule = {
+        'reset-demo': {
+            'task': 'miqa.core.tasks.reset_demo',
+            'schedule': timedelta(days=1),
+        }
+    }
```

Comment:
By placing this here, you're making this code execute at import time. Whenever possible, it'd be nice to avoid import side effects, though I realize Celery and Django both somewhat abuse this.

Also, I'd suggest using the Django settings registry as the source of truth for all Celery config, since that's exactly what
```python
config_source='django.conf:settings', namespace='CELERY'
```
is setting up. It also makes it easier to test and debug settings if you know they're all guaranteed to be defined at the same place. You'd just have to use the Django setting named as `CELERY_BEAT_SCHEDULE`.

If you want the value of this setting to dynamically depend on another setting, [Django Configurations allows the use of `@property`](https://django-configurations.readthedocs.io/en/stable/patterns.html#property-settings). You could do:
```python
class MiqaMixin(ConfigMixin):
    ...

    @property
    def CELERY_BEAT_SCHEDULE(self):
        if self.DEMO_MODE:
            return {
                'reset-demo': {
                    'task': 'miqa.core.tasks.reset_demo',
                    'schedule': timedelta(days=1),
                }
            }
        else:
            return {}
```

Alternatively, you could have something like:
```python
class MiqaMixin(ConfigMixin):
    ...

    CELERY_BEAT_SCHEDULE = {}

    @staticmethod
    def before_binding(configuration):
        if configuration.DEMO_MODE:
            configuration.CELERY_BEAT_SCHEDULE.update({
                'reset-demo': {
                    'task': 'miqa.core.tasks.reset_demo',
                    'schedule': timedelta(days=1),
                }
            })
```

which is a little more complicated, but more extensible.

Reply:
Changed in [5a36a70](https://github.com/OpenImaging/miqa/pull/635/commits/5a36a70f895942267b886b82c0ed7cabb62d4a76)